doc/YamlConfigFile.md

```diff
@@ -0,0 +1,298 @@
+# YAML configuration file
+
+This page describes the configuration file, which controls the behavior of the Windows service.
+
+You can find configuration file samples in the [examples](../examples) directory of the source code repository.
+Actual samples are being also published as a part of releases on GitHub and NuGet.
+
+## File structure
+
+YAML Configuration file shuold be in following format
+
+Example:
+
+```yaml
+id: jenkins
+name: Jenkins
+description: This service runs Jenkins continuous integration system.
+env:
+    - name: JENKINS_HOME
+      value: '%BASE%'
+executable: java
+arguments: >
+    -Xrs
+    -Xmx256m
+    -jar "%BASE%\jenkins.war"
+    --httpPort=8080
+log:
+    mode: roll
+```
+
+## Environment variable expansion
+
+Configuration YAML files can include environment variable expansions of the form `%Name%`. 
+Such occurrences, if found, will be automatically replaced by the actual values of the variables. [Read more. . . ](xmlConfigFile.md#environment-variable-expansion)
```

Comment:
Is it really supported at the moment? I do not see it in code though I am looking from phone and might be missing it

Reply:
I guess we are talking about [this](https://github.com/buddhikac96/winsw/blob/9dd7fb4d93055a582628ce8d963b1016d9715c38/src/Core/WinSWCore/Configuration/YamlConfiguration.cs#L500)??

Reply:
Nope. Syntax like `  <executable>%BASE%\myExecutable.exe</executable>` in XML when the variable is actually used in other fields. This is what `Environment variable expansion` means